### PR TITLE
Add a filter factory, so people can use FilterFactory instead of definin...

### PR DESCRIPTION
**Pull request by harikt**
...g the path to the scripts instace. Thank you v2 thoughts of @pmjones

### CODE REVIEW

**Comment by harikt** [comment]
I have rerun the failed issue in scrutinizer https://scrutinizer-ci.com/g/harikt/Aura.Filter/inspections/8bd58ee9-4f2a-42e4-a41e-03a9bf30ccab . This passes. 

@pmjones I wished if you merge this. And it will make something good.

Thanks

File: src/Aura/Filter/FilterFactory.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            new RuleLocator(array_merge(
+                $this->registry(),
+                ['any' => function () {
+                    $rule = new Rule\Any;
+                    $rule->setRuleLocator(new RuleLocator(
+                        $this->registry()
+                    ));
+
+                    return $rule;
+                }]
```
Comment (cordoval) [inline src/Aura/Filter/FilterFactory.php:48]: i think this indenting is wrong

File: src/Aura/Filter/FilterFactory.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                )
+            ),
+            $this->getTranslator()
+        );
+    }
+
+    public function getTranslator()
+    {
+        if (! $this->translator) {
+            $this->translator = new Translator(
```
Comment (cordoval) [inline src/Aura/Filter/FilterFactory.php:58]: this piece here was removed from the component, not sure why we want to put this back in?

**Comment by cordoval** [comment]
oh disregard my comments, i see this PR was for develop and not develop-2

**Comment by harikt** [comment]
@cordoval :)

**Pull request merged**